reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/Micrometer.java

```diff
@@ -93,6 +95,20 @@ public final class Micrometer {
 		return new MicrometerObservationListenerFactory<>(registry);
 	}
 
+	/**
+	 * Similar to {@link #observation(ObservationRegistry)} but enables providing
+	 * a function creating the Micrometer {@link Observation}
+	 * representing the runtime of the publisher to the provided {@link ObservationRegistry}.
```

Comment:
What is now missing IMO is the specification that when the supplier returns null, the behaviour is the same as the variant without a supplier and a default Observation is created.

Reply:
OK, I've added a sentence in the javadoc to explain the behavior if the function returns `null`

Reply:
Much appreciated!